components/builder-api/src/server/resources/profile.rs

```diff
@@ -170,6 +172,61 @@ async fn revoke_access_token(req: HttpRequest,
         Err(err) => return err.into(),
     };
 
+    let get_tokens_req = req.clone();
+
+    let tokens_response = get_access_tokens(get_tokens_req).await;
+    let response_body = tokens_response.into_body();
+
+    let response_bytes = match to_bytes(response_body).await {
+        Ok(bytes) => bytes,
+        Err(_) => {
+            let body = Bytes::from_static(b"Error reading access tokens response.");
+            return HttpResponse::with_body(StatusCode::INTERNAL_SERVER_ERROR, BoxBody::new(body));
+        }
+    };
+
+    let access_tokens_json: Value = match serde_json::from_slice(&response_bytes) {
+        Ok(json) => json,
+        Err(_) => {
+            let body = Bytes::from_static(b"Error parsing access tokens.");
+            return HttpResponse::with_body(StatusCode::NOT_ACCEPTABLE, BoxBody::new(body));
+        }
+    };
+
+    let access_tokens: Vec<Value> =
+        match access_tokens_json.get("tokens").and_then(|t| t.as_array()) {
+            Some(tokens) => tokens.clone(),
+            None => {
+                let body = Bytes::from_static(b"Tokens not present.");
+                return HttpResponse::with_body(StatusCode::NO_CONTENT, BoxBody::new(body));
+            }
+        };
+
+    let valid_token_id = match access_tokens.first() {
```

Comment:
As mentioned above, we are already getting the access tokens in this function so just move this below the call to `AccountToken::list`. Rather than just checking the first token, you should iterate over each token.